viewing_party/party.py

```diff
@@ -0,0 +1,140 @@
+def create_movie(movie_title, genre, rating):
+    new_movie = {
+        "title": movie_title,
+        "genre": genre,
+        "rating": rating
+    }
+
+    for key in new_movie:
+        if new_movie[key] == None:
+            return None
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, movie_title):
+    for movie in user_data["watchlist"]:
+        if movie_title == movie["title"]:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    
+    return user_data
+
+
+def get_watched_avg_rating(user_data):
+    avg_rating = 0.0
+    total_rating = 0.0
+    
+    for movie in user_data["watched"]:
+        total_rating += movie["rating"] 
+    
+    if len(user_data["watched"]) > 0:
+        avg_rating = total_rating/len(user_data["watched"])
+    return avg_rating
+
+
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+    
+    genre_dict = {}
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre in genre_dict:
+            genre_dict[genre] += 1
+        else:
+            genre_dict[genre] = 1
+
+    most_freq = genre
```

Comment:
It's a little surprising to see `most_freq` initialized from genre here. I had to look for a bit before I noticed that `genre` was created inside the previous for loop, and then I had to convince myself that `genre` could be guaranteed to always be initialized (which it is, since we never reach this part of the function if there are no movies, so we always pass through the loop at least once).

So this code is correct, but surprising for someone else to read (look at the several steps I needed to go through to understand it). Now imagine you've been away from this code for a while. Each time you or anyone else comes back to it you'll need to go through the same process to convince yourself that the code is correct. This is the kind of situation we should try to avoid when thinking about writing code for people rather than for computers. We should try to remove as much surprise from our code as we can.

Rather than initializing `most_freq` to the last genre we happened to see, another common pattern for something like this would be to set it to None at first, and create an additional variable that tracks the highest count we've seen (usually set to an impossibly low value, like -1). Then, as we iterate through the counts, we don't need to compare the current count to the count stored under the last found key, we can compare it to the last recorded max count, keeping those variables updated as we iterate.

Alternatively, we could use a series of function calls like `next(iter(genre_dict))` as a way to get the "first" key in the dictionary. This still depends on understanding the dictionary will be non-empty when we reach this point, but it avoids using a variable that only appears previously within a loop body.

A comment with a reminder could also be helpful, but in general, I find that if I'm leaving myself reminders in comments, that's a good hint that I should try to rewrite that code to be more understandable!